frontend/packages/ux-editor/src/components/Properties/PropertiesHeader/EditComponentIdRow/EditComponentIdRow.tsx

```diff
@@ -42,12 +47,26 @@ export const EditComponentIdRow = ({
     }
   };
 
+  const hasDuplicateDataType = (id: string) => {
+    if (
+      component.type === ComponentType.FileUpload ||
+      component.type === ComponentType.FileUploadWithTag
+    ) {
+      const dataTypeExists = appMetadata.dataTypes?.find((dataType) => dataType.id === id);
+      if (dataTypeExists) {
+        return t('ux_editor.modal_properties_component_id_not_unique_error');
+      }
```

Comment:
I suggest returning the boolean value here , considering the function is named `hasDuplicateDataType` and is used in an if statement:

```suggestion
      return appMetadata.dataTypes?.find((dataType) => dataType.id.toLowerCase() === id.toLowerCase());
```

Reply:
Nice, this is a lot better 👍